core/isolate.rs

```diff
@@ -233,16 +244,22 @@ impl Isolate {
       // return value.
       // TODO(ry) check that if JSError thrown during respond(), that it will be
       // picked up.
+      if !zero_copy_alloc_ptr.is_null() {
+        isolate.zero_copy_release(zero_copy_alloc_ptr);
+      }
```

Comment:
This seems to be fixing a fairly serious memory leak. I'm trying to think of a way we could have tested for this. Any suggestions? (I was thinking we could have a counter for every time a zero copy buf is dropped, and then check the metric in the unit tests below.)

Reply:
So although I did not add a "named test" for it, it definitely gets tested: the `CHECK(ref_count_map_.empty())` in `~ArrayBufferAllocator()` checks that, and it runs on every invocation of deno.
Adding another contrived test for it is totally pointless.

What we are missing is a way to "test the tests", it'd be nice if I could write a program that intentionally leaks a buffer and verify if that actually crashes. Any ideas where that would go?

Reply:
> What we are missing is a way to "test the tests", it'd be nice if I could write a program that intentionally leaks a buffer and verify if that actually crashes. Any ideas where that would go?

In Rust you can do `#[should_panic]` above a test...

Reply:
> In Rust you can do #[should_panic] above a test...

Unfortunately the check runs in the c++ destructor of a static variable, I don't think this would lead to a rust panic.
Rust would have already returned from `panic_unwind()` and (thus no longer able to catch panics) by the time that happens.